app/app/src/main/kotlin/com/hedvig/app/feature/settings/SettingsDestination.kt

```diff
@@ -0,0 +1,309 @@
+                    contentColor = MaterialTheme.colorScheme.primary,
+                  ),
+                  modifier = Modifier.fillMaxWidth(1f),
+                )
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+@Composable
+@HedvigPreview
+fun PreviewSettingsScreen() {
+  HedvigTheme(useNewColorScheme = true) {
+    Surface {
+      SettingsScreen(
+        onBackPressed = {},
+        notificationsEnabled = false,
+        onNotificationInfoDismissed = {},
+        onLanguageSelected = {},
+        onThemeSelected = {},
+        uiState = SettingsViewModel.SettingsUiState(
+          selectedLanguage = Language.SV_SE,
+          languageOptions = listOf(Language.SV_SE, Language.EN_SE),
+          selectedTheme = Theme.LIGHT,
+          themeOptions = listOf(Theme.LIGHT),
+          notificationOn = true,
+          showNotificationInfo = true,
+        ),
+      )
+    }
+  }
+}
+
+@Composable
+internal fun LanguageWithDialog(
+  languageOptions: List<Language>,
+  selectedLanguage: Language,
+  selectLanguage: (Language) -> Unit,
+  enabled: Boolean,
+) {
+  val context = LocalContext.current
+  var showLanguagePickerDialog by rememberSaveable { mutableStateOf(false) }
+  if (showLanguagePickerDialog) {
+    SingleSelectDialog(
+      title = stringResource(R.string.language_picker_modal_title),
+      optionsList = languageOptions,
+      onSelected = selectLanguage,
+      getDisplayText = { context.getString(it.getLabel()) },
+      getIsSelected = { selectedLanguage == it },
+      getId = { it.name },
+      onDismissRequest = { showLanguagePickerDialog = false },
+    )
+  }
+
+  HedvigBigCard(
+    onClick = { showLanguagePickerDialog = true },
+    hintText = stringResource(id = R.string.language_picker_modal_title),
+    inputText = context.getString(selectedLanguage.getLabel()),
+    enabled = enabled,
+    modifier = Modifier.fillMaxWidth(),
+  )
+}
+
+@Composable
+internal fun ThemeWithDialog(
+  themeOptions: List<Theme>,
+  selectedTheme: Theme,
+  selectTheme: (Theme) -> Unit,
+  enabled: Boolean,
+) {
+  val context = LocalContext.current
+  var showThemePickerDialog by rememberSaveable { mutableStateOf(false) }
+  if (showThemePickerDialog) {
+    SingleSelectDialog(
+      title = stringResource(R.string.SETTINGS_THEME_TITLE),
+      optionsList = themeOptions,
+      onSelected = selectTheme,
+      getDisplayText = { context.getString(it.getLabel()) },
+      getIsSelected = { selectedTheme == it },
+      getId = { it.name },
+      onDismissRequest = { showThemePickerDialog = false },
+    )
+  }
+
+  HedvigBigCard(
+    onClick = { showThemePickerDialog = true },
+    hintText = stringResource(R.string.SETTINGS_THEME_TITLE),
+    inputText = stringResource(id = selectedTheme.getLabel()),
+    enabled = enabled,
+    modifier = Modifier.fillMaxWidth(),
+  )
+}
+
+private fun startSettingsActivity(context: Context) {
```

Comment:
Hmm think it'd be a tiny bit nicer to name this something like `startAndroidSettingsActivity`?
At first glance I was wondering why our settings are an activity but I realize now this is for the android settings themselves, not an activity we control.
If you don't like the name feel free to not change it though, or if you have a better one just pick that one instead 😊

Reply:
Good idea!